npep/npep-137-conformance-profiles.md

```diff
@@ -0,0 +1,182 @@
+profile conformance.
+
+## Integration
+
+The conformance profile test suite can be integrated, invoked and run from your implementation
+using two methods:
+
+* The `go test` CLI commands specifying the required information need to generate
+a conformance test report. Sample:
+```
+go test  -v ./conformance -run TestConformanceProfiles -args --conformance-profiles=AdminNetworkPolicy,BaselineAdminNetworkPolicy --organization=ovn-org -project=ovn-kubernetes -url=<> -version=0.1.1 -contact=<>
+```
+* Using the conformance profile test suite `TestConformanceProfiles` by directly customizing it by providing
+the correct arguments. Sample:
+```
+cpSuite := suite.NewConformanceProfileTestSuite(
+            suite.ConformanceProfileOptions{
+                    suite.Options: cSuiteDefaultOptions,
+                    Implementation: confv1alpha1.Implementation{
+                            Organization: "ovn-org",
+                            Project:      "OVN-Kubernetes CNI",
+                            URL:          "https://github.com/ovn-org/ovn-kubernetes",
+                            Version:      "0.1.1",
+                            Contact:      []string{"@tssurya"},
+                    },
+                    ConformanceProfiles: sets.New(
+                            suite.ANPConformanceProfileName,
+                            suite.BANPConformanceProfileName,
+                    ),
+            })
+```
+
+## Reporting process and certification
+
+The reporting process is related to a specific API's version and channel (core and experimental).
+There are fields in the ConformanceReport CRD that inclues such information. Any implementation
+can run the existing conformance test suite specifying the profiles they support and that will
+generate an output that looks like this:
+
+```
+    Conformance report:
+        apiVersion: policy.networking.k8s.io/v1alpha1
+        date: "2023-10-03T08:15:25+02:00"
+        implementation:
+          contact:
+          - "@tssurya"
+          organization: ovn-org
+          project: OVN-Kubernetes CNI
+          url: "https://github.com/ovn-org/ovn-kubernetes"
+          version: 0.1.1
+        kind: ConformanceReport
+        networkPolicyV2APIVersion: v0.1.1
+        profiles:
+        - core:
+            failedTests:
+            - AdminNetworkPolicyIngressSCTP
+            - AdminNetworkPolicyEgressUDP
+            result: failure
+            statistics:
+              Failed: 2
+              Passed: 5
+              Skipped: 0
+            summary: ""
+          name: AdminNetworkPolicy
+        - core:
+            result: success
+            statistics:
+              Failed: 0
+              Passed: 7
+              Skipped: 0
+            summary: ""
+          name: BaselineAdminNetworkPolicy
+```
+
+This can then be uploaded to network-policy-api/conformance/reports/v.x.x/cni-name.yaml by
+opening a PR. That will then be reviwed and approved by mainainers thus recognising the
```

Comment:
I'm wondering if we should require implementations to also provide additional information in/alongside with this report, i.e. github action/jenkins or any other CI/CD job definition and run links. As network-policy-api maintainers, we might want to review how the conformance test was run to led to the generation of the report?

Reply:
oh great point! that way we are having something that maintainers can actually check and verify against. Let me add that.

Reply:
I am going to call this `VerificationLink`

Reply:
haha actually `additionalInfo` is best lol, we leave freedom to implementations what form of proof they want to provide